shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -568,14 +569,20 @@ fun NearbyStaticData.withRealtimeInfoWithoutTripHeadsigns(
     fun UpcomingTripsMap.maybeFilterCancellations(isSubway: Boolean) =
         if (filterCancellations) this.filterCancellations(isSubway) else this
 
-    fun RealtimePatterns.shouldShow(): Boolean {
+    fun RealtimePatterns.shouldShow(stopPatterns: NearbyStaticData.StopPatterns): Boolean {
         if (!allDataLoaded && showAllPatternsWhileLoading) return true
         val isUpcoming =
             when (cutoffTime) {
                 null -> this.isUpcoming()
                 else -> this.isUpcomingWithin(filterAtTime, cutoffTime)
             }
-        return (isTypical() || isUpcoming) && !isArrivalOnly()
+        val isLastStopOnRoutePattern =
+            stopPatterns.patterns.any {
+                val lastStopIdInPattern = it.stopIds.last()
+                val stop = stopPatterns.stop
+                lastStopIdInPattern == stop.id || stop.childStopIds.contains(lastStopIdInPattern)
+            }
```

Comment:
Sorry I missed this before!! I think part of the problem is that `stopPatterns.patterns` contains the patterns in *both* directions, whereas we only want to consider the patterns in the current direction being considered (In this example, looking only at the Northbound Oak Grove route pattern, not Southbound to Forest Hills).

`this.patterns` would be the better fit for that purpose. We can use the global data to resolve the ordered list of stops for each pattern in `this.patterns` `representativeTripId`